homeassistant/components/plugwise/sensor.py

```diff
@@ -0,0 +1,422 @@
+                    break
+
+    async_add_entities(entities, True)
+
+
+class PwThermostatSensor(SmileGateway, Entity):
+    """Thermostat (or generic) sensor entities."""
+
+    def __init__(self, api, coordinator, name, dev_id, sensor, sensor_type):
+        """Set up the Plugwise API."""
+        super().__init__(api, coordinator)
+
+        self._api = api
+        self._dev_id = dev_id
+        self._sensor_type = sensor_type
+        self._name = name
+        self._sensor = sensor
+        self._state = None
+        self._model = None
+        self._unit_of_measurement = None
+        if self._sensor_type is not None:
+            self._model = self._sensor_type[0]
+            self._unit_of_measurement = self._sensor_type[1]
+            self._dev_class = self._sensor_type[2]
+            self._icon = self._sensor_type[3]
+        else:
+            self._dev_class = "auxiliary"
+
+        self._heating_state = False
+        self._cooling_state = False
+
+        if self._dev_id == self._api.heater_id:
+            self._name = "Auxiliary"
+        sensorname = sensor.replace("_", " ").title()
+        self._sensorname = f"{self._name} {sensorname}"
+
+        if self._dev_id == self._api.gateway_id:
+            self._name = f"Smile {self._name}"
+
+        self._unique_id = f"{dev_id}-{sensor}"
+
+    @property
+    def device_class(self):
+        """Device class of this entity."""
+        return self._dev_class
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._sensorname
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_info(self) -> Dict[str, any]:
+        """Return the device information."""
+
+        device_information = {
+            "identifiers": {(DOMAIN, self._dev_id)},
+            "name": self._name,
+            "manufacturer": "Plugwise",
+        }
+
+        if self._model is not None:
+            device_information["model"] = self._model
+
+        if self._dev_id != self._api.gateway_id:
+            device_information["via_device"] = (DOMAIN, self._api.gateway_id)
+
+        return device_information
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon for the sensor."""
+        if self._sensor_type is None:
+            if self._heating_state:
+                return FLAME_ICON
+            if self._cooling_state:
+                return COOL_ICON
+            return IDLE_ICON
+        return self._icon
+
+    def _process_data(self):
+        """Update the entity."""
+        data = self._api.get_device_data(self._dev_id)
+
+        if not data:
+            _LOGGER.error("Received no data for device %s.", self._name)
```

Comment:
```suggestion
            _LOGGER.error("Received no data for device %s.", self._name)
            self.async_write_ha_state()
            return
```

Then you can shift the below block over

Reply:
Good catch, tnx.